Dockerfile.ubuntu20

```diff
@@ -53,11 +53,15 @@ RUN mkdir /tmp/build && \
     rm -r /tmp/build
 
 # HPC-X (2.14.0)
+# grep + sed is used as a workaround to update hardcoded pkg-config / libtools archive / CMake prefixes
 ENV HPCX_VERSION=2.14
 ENV HPCX_NCCL_VERSION=2.16
 RUN cd /tmp && \
-    wget -q -O - http://blobstore.s3.ord1.coreweave.com/drivers/hpcx-v${HPCX_VERSION}-gcc-MLNX_OFED_LINUX-5-ubuntu20.04-cuda11-gdrcopy2-nccl${HPCX_NCCL_VERSION}-x86_64.tbz | tar xjf - && \
-    mv hpcx-v${HPCX_VERSION}-gcc-MLNX_OFED_LINUX-5-ubuntu20.04-cuda11-gdrcopy2-nccl${HPCX_NCCL_VERSION}-x86_64 /opt/hpcx
+    export HPCX_DISTRIBUTION="hpcx-v${HPCX_VERSION}-gcc-MLNX_OFED_LINUX-5-ubuntu20.04-cuda11-gdrcopy2-nccl${HPCX_NCCL_VERSION}-x86_64" \
+           HPCX_DIR="/opt/hpcx" && \
+    wget -q -O - http://blobstore.s3.ord1.coreweave.com/drivers/${HPCX_DISTRIBUTION}.tbz | tar xjf - && \
+    grep -IrlF "/build-result/${HPCX_DISTRIBUTION}" ${HPCX_DISTRIBUTION} | xargs -rd'\n' sed -i -e "s:/build-result/${HPCX_DISTRIBUTION}:${HPCX_DIR}:g" && \
```

Comment:
Is this acting on every single file inside the hpc-x distribution? the commandline is a  bit hard for me to parse.

Reply:
Everything but binary files, yep. To break it down:
```sh
grep -IrlF "/build-result/${HPCX_DISTRIBUTION}" ${HPCX_DISTRIBUTION} \
| xargs -rd'\n' \
sed -i -e "s:/build-result/${HPCX_DISTRIBUTION}:${HPCX_DIR}:g"
```
1. `grep -IrlF` = Recursively (`r`) search every text (`I`) file in the distribution and list the names (`l`) of those that contain the exact string (`F`) "`/build-result/${HPCX_DISTRIBUTION}`" (which expands to look like "`/build-result/hpcx-v2.14-gcc-MLNX_OFED_LINUX-5-ubuntu20.04-cuda11-gdrcopy2-nccl2.16-x86_64`"), then
2. `xargs -rd'\n'` = For each of those files in `grep`'s newline-delimited (`d'\n'`) output, if there are any (`r`),
3. `sed -i -e "s:/build-result/${HPCX_DISTRIBUTION}:${HPCX_DIR}:g"` = Perform an inplace (`i`) find-and-replace (`-e s: : :g`), replacing that string `/build-result/${HPCX_DISTRIBUTION}` in the matched files with `${HPCX_DIR}`.

There are binary files which match the pattern too somewhere among their bytes, so `grep -I` filters those out beforehand to not accidentally mangle them in any way. The list of files that actually end up matched by the `grep` call is almost exactly the long list in my first PR comment (but before they are `mv`ed over to `/opt/hpcx`).

Caveats: `sed` has no literal strings mode so it is interpreted as a `sed` regex (which is very limited, it only affects `.` here), but it is such a long and specific string it won't actually match anything else anyway.

Reply:
How long does this step take?

Reply:
About one second or less, usually.